inbm/dispatcher-agent/dispatcher/dispatcher_class.py

```diff
@@ -92,7 +92,7 @@ def __init__(self, args: list[str], broker: DispatcherBroker, install_check_serv
         self._dispatcher_broker = broker
         self._install_check_service = install_check_service
         self.update_queue: Queue[Tuple[str, str, Optional[str]]] = Queue(1)
-        self._thread_count = 1
+        self._thread_count = 2
```

Comment:
we will need to have a different mechanism to synchronize individual tasks--we can't change this from 1->2 here as we will have overlapping manifests running at the same time (like SOTA+FOTA) which will cause problems. maybe each manifest/task can signal that it's done by updating the database, so the scheduler knows it's OK to submit the next task?

Reply:
The root cause of the thread increment is because of the apscheduler. The job thread only will be created after a job being added. Hence it increases the active thread count. 
The job list is empty while creating the apscheduler. The apscheduler doesn't activate the job thread at the beginning.  This causes the active_start_count to miss out on the apscheduler thread. 
To fix the issue, I created a dummy job to print the message when starting the apscheduler. The message will print out after 1s. This activates the apscheduler to create the job thread, and this thread is counted into active_start_count now.